addons/mail/static/src/models/thread_viewer/thread_viewer.js

```diff
@@ -218,6 +231,15 @@ function factory(dependencies) {
         componentHintList: attr({
             default: [],
         }),
+        composer: many2one('mail.composer', {
+            related: 'thread.composer',
+        }),
+        composerHasFocus: attr({
+            related: 'composer.hasFocus',
+        }),
+        lastMessageIsVisible: attr({
```

Comment:
Mmmh I think I see your point but we also need the fact that the message has been shown ... Are you saying that this reference is set only on last message that have been "visible" ? So we need to check in message_list if the reference to the last message is the same message than the last "shown" in thread viewer ?

Reply:
Not sure if there are some technical flaws with how I see it, but here it is:

I think component message list should be as dumb as possible. IMO it should only tell "these messages are visible" to the thread viewer, and the thread viewer itself auto-deduce seen last message of current partner from that.

Reply:
I understand but I think it's technically heavier as for the moment message list just look at the last message and don't bother with other. If you do that, then all messages should be tracked to know if they are visible or not and then send the list of {message/visible_or_not_boolean} updated at some point to the thread_viewer. For now message_list just say "ok the last I have is seen or not" which is lighter but has the drawback to make message list a little bit more "intelligent" by keeping track of the last message it has which is not that much of intelligence I think ..

Reply:
This is a rough idea of how I think it should work in order to be super easy to understand and super flexible to change of any kind.

If technical processing is a concern, we could limit it to component just notifying the last visible message in message list when it becomes visible.

This would basically have the same technical processing cost as your solution while having the nice aspects I just mentioned in the beginning of this post.

Reply:
I don't see/understand the difference(s) between your "limiting" solution and the one actually written ?

Reply:
When most recent message is partially visible, it just calls a public method on thread viewer with message as parameter. Thread viewer itself update `seenMessage` if necessary.